tests/upgrades/test_capsule.py

```diff
@@ -112,57 +111,51 @@ def test_post_user_scenario_capsule_sync(self):
             2. The repos/rpms from satellite should be synced to satellite
 
         """
-        env_name = get_entity_data(self.__class__.__name__)['env_name']
-        org_name = entities.Organization().search(query={'search': f'id={self.org_id}'})[0].label
-        capsule = entities.SmartProxy().search(query={'search': f'name={self.cap_host}'})[0]
+        request.addfinalizer(lambda: __cleanup(content_view, repo, product))
```

Comment:
Does the lambda work correctly here even though `content_view`, `repo`, and `product` aren't yet defined?

Reply:
My intuition leans heavily toward it not working, but I'd like to see it tested and verified that those entities are properly cleaned up.

Reply:
@mshriver @JacobCallahan It works here, I have re-executed the test cases multiple times, and no extra cleanup has required for that. here we are calling the function and that is already defined, the order only matters when we call some method that is not defined.

In the below example if we call ```request.addfinalizer(org.delete)``` before the ```org``` definition then it will get failed because we don't have any ```org``` definition,  So in this case, we need to call the ```request.addfinalizer``` after the definition.

org = entities.Organization().search(query={'search': f'name={pre_test_name}_org'})[0]
request.addfinalizer(org.delete)